accounts-db/src/tiered_storage/meta.rs

```diff
@@ -47,8 +67,18 @@ pub trait TieredAccountMeta: Sized {
     /// meta.
     fn with_flags(self, flags: &AccountMetaFlags) -> Self;
 
-    /// Returns the balance of the lamports associated with the account.
-    fn lamports(&self) -> u64;
+    /// Whether the account has zero lamports.
+    fn has_zero_lamports(&self) -> bool;
+
+    /// Returns the balance of the lamports associated with the account
+    /// from the TieredAccountMeta, or None if the lamports is stored
+    /// inside the optional field.
+    fn lamports_from_meta(&self) -> Option<u64>;
+
+    /// Returns the balance of the lamports associated with the account
+    /// from the optional fields, or None if the lamports is stored
+    /// inside the TieredAccountMeta.
+    fn lamports_from_optional_fields(&self, _account_block: &[u8]) -> Option<u64>;
```

Comment:
Here's the API change for the TieredStorageMeta, and TieredReadableAccount will use these APIs to implement its ReadableAccount::lamport() function.

Reply:
> I think the TieredAccountMeta trait should stick with just lamports(), and let the HotPacked Meta implementation handle where to read from.

It is because HotAccountMeta alone cannot provide the full picture because it does not possess the data.

One solution is to convert the current HotAccountMeta to HotAccountStoredMeta, and create a new HotAccountMeta that contains everything.

(Sounds like this would be a separate PR and this PR on top of it?)

Reply:
Now they belong to HotAccount.